src/NuGet.Services.Metadata.Catalog.Monitoring/Status/PackageMonitoringStatusService.cs

```diff
@@ -28,40 +26,74 @@ public class PackageMonitoringStatusService : IPackageMonitoringStatusService
         /// </summary>
         private IStorageFactory _storageFactory;
 
-        /// <summary>
-        /// The <see cref="JsonSerializerSettings"/> to use to save and load statuses of packages.
-        /// </summary>
-        private static JsonSerializerSettings SerializerSettings => _serializerSettings.Value;
-        private static Lazy<JsonSerializerSettings> _serializerSettings = new Lazy<JsonSerializerSettings>(() =>
+        public PackageMonitoringStatusService(IStorageFactory storageFactory, ILogger<PackageMonitoringStatusService> logger)
         {
-            var settings = new JsonSerializerSettings();
+            _logger = logger;
+            _storageFactory = storageFactory;
+        }
 
-            settings.Converters.Add(new NuGetVersionConverter());
-            settings.Converters.Add(new StringEnumConverter());
+        public async Task<IEnumerable<PackageMonitoringStatusListItem>> ListAsync(CancellationToken token)
+        {
+            var listTasks =
+                Enum.GetNames(typeof(PackageState))
+                .Select(state => GetListItems(state, token))
+                .ToList();
 
-            return settings;
-        });
+            return
+                (await Task.WhenAll(listTasks))
+                .Where(list => list != null && list.Any())
+                .Aggregate((current, next) => current.Concat(next))
+                .Where(i => i != null);
+        }
 
-        public PackageMonitoringStatusService(IStorageFactory storageFactory, ILogger<PackageMonitoringStatusService> logger)
+        private async Task<IEnumerable<PackageMonitoringStatusListItem>> GetListItems(string state, CancellationToken token)
         {
-            _logger = logger;
-            _storageFactory = storageFactory;
+            await Task.Yield();
```

Comment:
>Yield [](start = 23, length = 5)

There should be no need for the `Task.Yield` since you await on `storage.List` anyways, right?

Reply:
yup, nice catch